tests/ui/unwrap.rs

```diff
@@ -10,7 +10,23 @@ fn unwrap_result() {
     let _ = res.unwrap();
 }
 
+// Should not get linted.
+#[test]
+fn with_test() {
+    let opt = Some(0);
+    let _ = opt.unwrap();
+}
+
+// Should not get linted.
+#[cfg(test)]
+fn with_cfg_test() {
+    let res: Result<u8, ()> = Ok(0);
+    let _ = res.unwrap();
+}
+
 fn main() {
     unwrap_option();
     unwrap_result();
+    with_test();
+    with_cfg_test();
```

Comment:
I don't think you need to add these function to main.
Use https://rustc-dev-guide.rust-lang.org/tests/adding.html#revisions with compile flag for test
is better I think.

Reply:
Thank you for the review and suggestion. I will fix this later.